### PR TITLE
Add test data support, generic query and more tests

### PR DESCRIPTION
**Pull request by pakrym**
And samples.

### CODE REVIEW

File: sdk/monitor/Azure.Monitory.Query/src/Models/LogsQueryResultRow.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+        internal LogsQueryResultRow(Dictionary<string, int> columns, JsonElement row)
+        {
+            _columns = columns;
+            _row = row;
+        }
+
+        public int Count => _row.GetArrayLength();
+
+        public int GetInt32(int index) => _row[index].GetInt32();
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitory.Query/src/Models/LogsQueryResultRow.cs:23]: Non-boxing overloads

File: sdk/monitor/Azure.Monitory.Query/tests/LogsQueryClientClientLiveTests.cs [hunk 2]
```diff
@@ -30,11 +38,103 @@ public async Task CanQuery()
...
+            Assert.AreEqual(1, resultTable.Rows[0].GetInt32(LogsTestData.IntColumnName));
+
+            Assert.AreEqual(false, resultTable.Rows[0].GetBoolean(2));
+            Assert.AreEqual(false, resultTable.Rows[0].GetBoolean(LogsTestData.BoolColumnName));
+
+            Assert.AreEqual(0f, resultTable.Rows[0].GetSingle(3));
+            Assert.AreEqual(0f, resultTable.Rows[0].GetSingle(LogsTestData.FloatColumnName));
+        }
+
+        [Test]
```
Comment (christothes) [inline sdk/monitor/Azure.Monitory.Query/tests/LogsQueryClientClientLiveTests.cs:62]: `[RecordedTest]`?

File: sdk/monitor/Azure.Monitory.Query/src/Models/LogsQueryResultRow.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace Azure.Monitory.Query.Models
+{
+    public class LogsQueryResultRow
```
Comment (christothes) [inline sdk/monitor/Azure.Monitory.Query/src/Models/LogsQueryResultRow.cs:10]: docs?

File: sdk/monitor/Azure.Monitory.Query/src/Models/LogsQueryResultRow.cs (hunk 1 above)
Comment (christothes) [inline sdk/monitor/Azure.Monitory.Query/src/Models/LogsQueryResultRow.cs:23]: It's unclear what these methods do based on their names. Maybe something like `GetInt32ByIndex(int columnIndex)` and `GetInt32ByName(string columnName)`? 

Also, could these be generic? (`GetByIndex<T>(int columnIndex)` and `GetByName<T>(string columnName)`)

**Review by christothes — APPROVED** [review]

File: sdk/monitor/Azure.Monitory.Query/src/Models/LogsQueryResultRow.cs (hunk 1 above)
Comment (pakrym) [reply sdk/monitor/Azure.Monitory.Query/src/Models/LogsQueryResultRow.cs:23]: I was mimicking `DataReader` and `JsonElement` method here:
 https://docs.microsoft.com/en-us/dotnet/api/system.data.datatablereader.getint64?view=net-5.0
https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.getint32?view=net-5.0

File: sdk/monitor/Azure.Monitory.Query/src/Models/LogsQueryResultRow.cs (hunk 3 above)
Comment (pakrym) [reply sdk/monitor/Azure.Monitory.Query/src/Models/LogsQueryResultRow.cs:10]: To come. I want the feature team to take a look an agree on the direction.

File: sdk/monitor/Azure.Monitory.Query/tests/LogSenderClient.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            _workspaceId = workspaceId;
+            _pipeline = HttpPipelineBuilder.Build(options, new SignaturePolicy(workspaceId, sharedKey));
+            _clientDiagnostics = new ClientDiagnostics(options);
+        }
+
+        public async Task<Response> SendAsync(string tableName, IEnumerable<IDictionary<string, object>> values)
+        {
+            var data = JsonSerializer.Serialize(values);
+            var request = _pipeline.CreateRequest();
+            request.Uri.Reset(new Uri($"https://{_workspaceId}.ods.opinsights.azure.com/api/logs?api-version=2016-04-01"));
```
[the remark this answers is not in the document]
Comment (pakrym) [reply sdk/monitor/Azure.Monitory.Query/tests/LogSenderClient.cs:33]: This is a test utility client so I'm not sure how much the `ServiceVersion` matters here. Good point on the suffix.

File: sdk/monitor/Azure.Monitory.Query/src/Models/LogsQueryResultRow.cs (hunk 1 above)
Comment (pakrym) [reply sdk/monitor/Azure.Monitory.Query/src/Models/LogsQueryResultRow.cs:23]: Hm, I think most of the time the access pattern would be 1x per value and materializing ints is non-allocating and cheap. The only thing that might be worth caching are strings but I'm worried that allocating another array per row might hurt more than help.

**Pull request merged**